site/content/docs/Advanced/Ransomware Protection/_index.md

```diff
@@ -0,0 +1,86 @@
+    * There is no need to set a default Retention time
+  * Optional: Create a restricted access key as instructed above for additional protection
+  * Create a new Kopia repository
+    * Run: `kopia repo create s3 --bucket <bucket name> --access-key=<access key> --secret-access-key=<secret access key> --retention-mode COMPLIANCE --retention-period <retention time>`
+      * `retention-period`` can be specified in days (for instance 30d)
+  * After each backup (or at least before the retirement period is up) it is important to reset the Object Lock retention time on all current files.
+    * Kopia does not currently provide his capability, but the process is to:
+      * List all filenames in the bucket (assuming a single repo per bucket or at least consistent retention times)
+      * For each found file starting with 'x', 'q', or 'p': apply a PutObjectRetention command with the updated retention time.
+        * files starting with _log probably do not need additional retention applied
+      * This can be achieved using a reasonably simple python script `TBD`
+
+### How to restore a snapshot that was deleted by Ransomware (or some other process)
+  * If you have data that needs to be restored, make sure that either your retention time will not expire or your lifecycle data expiration is sufficient to ensure you can download your data before the cloud provider could remove it
+  * Disconnect the repo in Kopia
+  * Reconnect the repo in Kopia using the `--point-in-time` option (ex: --point-in-time=2021-11-29T01:10:00.000Z`)
+    * This option is only currently available when using an 's3' repo
+
+### A caveat about Ransomware protection
+
+Ransomware can be very sophisticated software.  It may run on your system for weeks, slowly encrypting data that is not frequently used, and only hitting actively used data last (to avoid detection for as long as possible).  This means that you should ensure your data retention period is long enough that you will be able to recover your data.  It also means that recovery may not be easy.  You may have many snapshots after the Ransomware has started modifying your system, and you may have made changes throughout that time such that there is no single snapshot that represents a good state of all of your files.  Restoration is likely to be a time-consuming task, and your best bet is to protect your system from these threats so they don't occur in the first place.  But if they do, Kopia can provide some additional security to protect your data.
+
+Additionally note that ransomware could theoretically weaponize object-locks to cost you a lot of money.  Becuase object-locks cannot be shortened are removed, Sufficiently malicious ransomware could upload large amounts of data and set a very long object lock that would make it impossible to delete.  It is strongly recommended to ensure you have appropriate quotas/limits on your buckets to limit potential storage costs.
```

Comment:
...`cannot be shortened are removed` --> `cannot be shortened or removed`

`removed, Sufficiently malicious` --> `removed, sufficiently malicious`